pkg/reconciler/messaging/channel/channel.go

```diff
@@ -168,7 +170,7 @@ func (r *Reconciler) syncSubscribers(ctx context.Context, channel *v1alpha1.Chan
 			ServiceAccount: channel.Spec.GoogleServiceAccount,
 			Secret:         channel.Spec.Secret,
 			Labels:         resources.GetPullSubscriptionLabels(controllerAgentName, channel.Name, genName, string(channel.UID)),
-			Annotations:    resources.GetPullSubscriptionAnnotations(channel.Name),
+			Annotations:    resources.GetPullSubscriptionAnnotations(channel.Name, clusterName),
```

Comment:
Why is it necessary for the channel to set the cluster name? Wouldn't we expect this to happen in the webhook? Is it so that if the channel has a `cluster-name` value different from the default, its topics and pull subscriptions get the same value?

If so, seems like other resources that also create PullSubscriptions like CloudPubSubSource should have the same behavior. That can be added in a follow-up PR.

Reply:
Yes, for WI, source and channel call Identity Reconciler function in there own reconciler, and create the KSA. (topic and pullsubscription reconciler only call Identity Reconciler function if the topic/pullsubscription obj doesn't have owner, which means they are created along without any up-level resources), and we pass the cluster annotation all the way down to publisher/receive-adaptor pod to make the KSA name to put into pod's serviceAccountName. This annotations setting here ensures that the KSA which reconcilers make has the same name as the topic/pullsubscription make.

Reply:
For sources, they use the different pullsubscription/topic resources from channel. And pullsubscription resource for source already include this annotation arg.(I guess it is used for the autoscaling annotation before.)